### PR TITLE
enable customized default location

### PR DESCRIPTION
**Pull request by mpwang**
Feature: Enable customization of default store files location using `AIVOC_HOME` environment variable
-----------------------------------------------------------------------------------------------------

### Motivation

I use [a-shell](https://github.com/holzschu/a-shell) to run Aivoc on my iPhone. However, due to file permission limitations, the `~/.aivoc_db` directory is expanded to `/private/var/mobile/Containers/Data/Application/xxxx/.aivoc_db`, which a-shell does not have permission to write in.

### Solution

To overcome this limitation, I propose providing a way to customize the default location of store files by introducing the `AIVOC_HOME` environment variable. This feature will allow users to set the `AIVOC_HOME` variable to a directory where Aivoc can write store files.

For example:

```bash
export AIVOC_HOME=$HOME/Documents
aivoc
```

This solution will be particularly useful for iOS users who cannot write in the `~` directory and can only write in `~/Documents/`, `~/Library/` and `~/tmp` directories.

As per a-shell documentation:

> In iOS, you cannot write in the ~ directory, only in ~/Documents/, ~/Library/ and ~/tmp. Most Unix programs assume the configuration files are in $HOME.

_pull request message written by ChatGPT_

### CODE REVIEW

File: voc_builder/config.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
 """Store configuration for project"""
+import os
 from pathlib import Path
 
 # Testings should patch these variables
 #
 # The default path for storing the vocabulary book
-DEFAULT_CSV_FILE_PATH = Path('~/aivoc_builder.csv').expanduser()
+aivoc_home = os.environ.get("AIVOC_HOME", "~")
```
Comment (piglei) [inline voc_builder/config.py:8]: Please rename the name of this config entry to `AIVOC_DATA_DIR`. Reason: There will be more config entries in the furture, `DATA_DIR` more precise than `HOME`.

**Comment by mpwang** [comment]
update as suggested, also add descprtion

File: voc_builder/config.py [hunk 2]
```diff
@@ -1,9 +1,14 @@
 """Store configuration for project"""
+import os
 from pathlib import Path
 
 # Testings should patch these variables
 #
 # The default path for storing the vocabulary book
-DEFAULT_CSV_FILE_PATH = Path('~/aivoc_builder.csv').expanduser()
+data_dir = os.environ.get('AIVOC_DATA_DIR', '~')
+if data_dir.endswith('/'):
```
Comment (piglei) [inline voc_builder/config.py:9]: 可以用 `Path(...) / 'subdir'` 的方式来拼接目录，效果类似于 `os.path.join()`，这样的话，这里的字符串处理逻辑可以去掉。

ref：https://docs.python.org/3/library/pathlib.html#operators

Comment (mpwang) [reply voc_builder/config.py]: thanks for the tips! unit test also included.

**Review by piglei — APPROVED** [review]

**Comment by piglei** [comment]
Thank you.

**Pull request merged**